src/components/Dropdown/Dropdown.Props.ts

```diff
@@ -58,4 +58,9 @@ export interface IDropdownOption {
    * Whether this option is currently selected.
    */
   isSelected?: boolean;
+
+  /**
+   * Optional custom renderer for the dropdown item
+   */
+  onRender?: (item: IDropdownOption) => JSX.Element;

```

Comment:
Yeah that makes sense, i was going to bring that up. I think it suffers from the same problem even and probably needs a deprecation for that and a better way to do custom stuff.

BTW, there is a cleaner approach here;
```
import { IRenderFunction } from '../../common/IRenderFunction';

onRenderItem?: IRenderFunction<IDropdownOption>;
```

Instead of just passing in the props, you should consider also passing in the default function. This lets the caller implement it like so:

```
onRenderItem: (props, defaultRender) => <div><img src='' />{ defaultRender(props) }</div>
```

Sometimes you want to just provide your own rendering, sometimes you want to wrap the original stuff inside of some composite.